### PR TITLE
#2850 Remove --cleanuplegacyfadersettings command line option

### PR DESCRIPTION
**Pull request by pljones**
**Short description of changes**

Removes undocumented `--cleanuplegacyfadersettings` command line option.

CHANGELOG: SKIP

**Context: Fixes an issue?**

Fixes: #2850

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

Tested

1.  Option no longer available:
```
$ ./Jamulus --cleanuplegacyfadersettings
./Jamulus: Unknown option '--cleanuplegacyfadersettings' -- use '--help' for help
```

2. `./Jamulus` after local build:

Before run:
```
 <storedfadertag0_base64>Tm8gTmFtZQ==</storedfadertag0_base64>
 <storedfadertag1_base64>QXllIE1pbiBIYW4=</storedfadertag1_base64>
 <storedfadertag2_base64>RGFuaWVsIEh0b28=</storedfadertag2_base64>
 <storedfadertag3_base64>SGFu</storedfadertag3_base64>
 <storedfadertag4_base64></storedfadertag4_base64>
 <storedfadertag5_base64></storedfadertag5_base64>
 <storedfadertag6_base64></storedfadertag6_base64>
```
Connected to a server, exited.

After run:
```
 <storedfadertag0_base64>Tm8gTmFtZQ==</storedfadertag0_base64>
 <storedfadertag1_base64>QXllIE1pbiBIYW4=</storedfadertag1_base64>
 <storedfadertag2_base64>RGFuaWVsIEh0b28=</storedfadertag2_base64>
 <storedfadertag3_base64>SGFu</storedfadertag3_base64>
 <storedfadertag4_base64></storedfadertag4_base64>
 <storedfadertag5_base64></storedfadertag5_base64>
 <storedfadertag6_base64></storedfadertag6_base64>
```

**What is missing until this pull request can be merged?**

Code review - deleting previously added code.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

**Comment by ann0see** [comment]
This should be documented, so no SKIP?

**Comment by pljones** [comment]
I'd rather it remained undocumented.  In #2839, the change log was
```
CHANGELOG: Client: Bugfix: Remove channel number from saved fader tag when --ctrlmidich is in use. In this release --cleanuplegacyfadersettings can be used on any saved fader settings that have been corrupted to attempt recovery.
```
-- the `--cleanuplegacyfadersettings` was documented _there_ as "in this release" (3.9.1).  No one should be expecting to work in 3.10.0, and certainly not 3.11.0.

**Comment by ann0see** [comment]
> No one should be expecting to work in 3.10.0, and certainly not 3.11.0.

This implies we should also remove it from the website.

**Comment by pljones** [comment]
> This implies we should also remove it from the website.

It was meant to be an undocumented option.  There shouldn't be any mention of it other than discussions / issues.

**Comment by pljones** [comment]
I'm seeing lots of
```
C:\Qt\6.5.2\msvc2019_64\include\QtGui/qtextoption.h(118): fatal error C1060: compiler is out of heap space
```

**Comment by ann0see** [comment]
Yes. I think I also saw that.

File: src/settings.cpp [hunk 1]
```diff
@@ -232,8 +232,6 @@ void CClientSettings::ReadSettingsFromXML ( const QDomDocument& IniXMLDocument,
     int  iValue;
     bool bValue;
 
-    bCleanUpLegacyFaderSettings = CommandLineOptions.contains ( "--cleanuplegacyfadersettings" );
```
Comment (ann0see) [inline src/settings.cpp:235]: Is passing ComandLineOptions still needed?

Otherwiese we don't fully undo https://github.com/jamulussoftware/jamulus/pull/2839/files

Comment (pljones) [reply src/settings.cpp:235]: It's used in `ReadSettingsFromXML` in CSettings generally - it should be issuing an unused variable warning...  Hm.

Comment (ann0see) [reply src/settings.cpp:235]: Are you sure? I don't find it in this function anymore.

Comment (pljones) [reply src/settings.cpp:235]:
```
src/settings.cpp:36:    ReadSettingsFromXML ( IniXMLDocument, CommandLineOptions );
```

**Review by ann0see — APPROVED** [review]
Ok. Then approving.

**Pull request merged**